<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
<title>BAli-Phy: MCMC Post-hoc Analysis: 89 sequences</title>
    <style type="text/css">
      body {margin:0; padding:0}
      ol li {padding-bottom:0.5em}

      th {padding-left: 0.5em; padding-right:0.5em}
      td {padding: 0.1em;}
      .backlit2 td {padding-left: 0.5em;}
      .backlit2 td {padding-right: 1.0em;}

#topbar {
	background-color: rgb(201,217,233);
	margin: 0;
	padding: 0.5em;
	display: table;
        width: 100%;
        position: fixed;
        top: 0;
}

#topbar #menu {
//	font-size: 90%;
	display: table-cell;
	text-align: right; 
        // Leave space for the menubar
        padding-right: 0.75em;
}
#topbar #path {
	font-weight: bold;
	display: table-cell;
	text-align: left; 
	white-space: nowrap;
}

      .content {margin:1em; margin-top: 3em}
      .backlit td {background: rgb(220,220,220);}

      :target:before {
        content:"";
        display:block;
        height:2em; /* fixed header height*/
        margin:-2em 0 0; /* negative fixed header height */
      }

      h1 {font-size: 150%;}
      h2 {font-size: 130%; margin-top:2.5em; margin-bottom: 1em}
      h3 {font-size: 110%; margin-bottom: 0.2em}// margin-top: 0.3em}

      ul {margin-top: 0.4em;}

      .center td {text-align: center};

      *[title] {cursor:help;}
      a[title] {vertical-align:top;color:#00f;font-size:70%; border-bottom:1px dotted;}
      .floating_picture {float:left;padding-left:0.5em;padding-right:0.5em;margin:0.5em;}    
      .r_floating_picture {float:right;padding-left:0.5em;padding-right:0.5em;margin:0.5em;}    
      table.backlit2 {border-collapse: collapse; }
      table.backlit2 tr:nth-child(odd) td { background-color: #F0F0F0; margin:0}
      .clear {clear:both;}    

      .modelname {font-weight: bold; color: blue}
      table.model {margin-left: 2em}
      table.model td {vertical-align: top}
      table.models td {vertical-align: top}

    </style>
</head>
<body>
<p id="topbar"> 
   <span id="path">Sections:</span>
   <span id="menu">
    [<a href="#data">Data+Model</a>]
    [<a href="#parameters">Parameters</a>]
    [<a href="#topology">Phylogeny</a>]
    [<a href="#alignment">Alignment</a>]
    [<a href="#mixing">Mixing</a>]
    [<a href="#analysis">Analysis</a>]
    [<a href="#models">Models+Priors</a>]
   </span>
</p>
<div class="content">
<h1>MCMC Post-hoc Analysis: 89 sequences</h1>
<h2 style="clear:both"><a class="anchor" name="data"></a>Data &amp; Model</h2>
<table class="backlit2">
<tr><th>Partition</th><th>Sequences</th><th>Lengths</th><th>Alphabet</th><th>Substitution&nbsp;Model</th><th>Indel&nbsp;Model</th><th>Scale&nbsp;Model</th></tr>
<tr>
 <td>1</td>
 <td>data/refcov89_ORF8_aln.fasta</td>
<td>234 - 303</td>
<td>DNA</td><td><a href="#S1">S1</a> = tn93</td>
 <td><a href="#I1">I1</a> = rs07</td>
 <td><a href="#scale1">scale1</a> ~ gamma[0.5,2]</td>
</tr>
</table>
<h2 class="clear"><a class="anchor" name="parameters"></a>Scalar variables</h2>
<table class="backlit2">
<tr><th>Statistic</th><th>Median</th><th title="95% Bayesian Credible Interval">95% BCI</th><th title="Auto-Correlation Time">ACT</th><th title="Effective Sample Size">ESS</th><th>burnin</th><th title="Potential Scale Reduction Factor based on width of 80% credible interval">PSRF-CI80%</th><th>PSRF-RCF</th></tr>
<tr >
<td>prior</td>
<td>392.7 </td>
<td>(335.6, 441.4)</td>
<td >122.2  </td>
<td  style="color:red">96</td>
<td >108
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>likelihood</td>
<td>-1.125e+04 </td>
<td>(-1.128e+04, -1.122e+04)</td>
<td >55  </td>
<td  style="color:orange">213</td>
<td >44
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>posterior</td>
<td>-1.086e+04 </td>
<td>(-1.091e+04, -1.081e+04)</td>
<td >40.49  </td>
<td  style="color:orange">290</td>
<td >108
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>prior_A</td>
<td>-298.1 </td>
<td>(-338.3, -266.2)</td>
<td >150.8  </td>
<td  style="color:red">77</td>
<td >689
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>|A|</td>
<td>383 </td>
<td>(354, 415)</td>
<td >20.02  </td>
<td >587</td>
<td >470</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>#indels</td>
<td>26 </td>
<td>(23, 30)</td>
<td  style="color:red">170.8  </td>
<td  style="color:red">68</td>
<td >601</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>|indels|</td>
<td>233 </td>
<td>(187, 285)</td>
<td >8.3  </td>
<td >1416</td>
<td >120</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>#substs</td>
<td>2517 </td>
<td>(2492, 2538)</td>
<td >60.64  </td>
<td  style="color:orange">193</td>
<td >120</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>P1/likelihood</td>
<td>-1.125e+04 </td>
<td>(-1.128e+04, -1.122e+04)</td>
<td >55  </td>
<td  style="color:orange">213</td>
<td >44
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>P1/prior_A</td>
<td>-298.1 </td>
<td>(-338.3, -266.2)</td>
<td >150.8  </td>
<td  style="color:red">77</td>
<td >689
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>P1/|A|</td>
<td>383 </td>
<td>(354, 415)</td>
<td >20.02  </td>
<td >587</td>
<td >470</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>P1/#indels</td>
<td>26 </td>
<td>(23, 30)</td>
<td  style="color:red">170.8  </td>
<td  style="color:red">68</td>
<td >601</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>P1/|indels|</td>
<td>233 </td>
<td>(187, 285)</td>
<td >8.3  </td>
<td >1416</td>
<td >120</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>P1/#substs</td>
<td>2517 </td>
<td>(2492, 2538)</td>
<td >60.64  </td>
<td  style="color:orange">193</td>
<td >120</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>Scale[1]*|T|</td>
<td>12.38 </td>
<td>(11.82, 12.94)</td>
<td >2.244  </td>
<td >5238</td>
<td >31
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>|T|</td>
<td>1.056 </td>
<td>(0.8457, 1.264)</td>
<td >1  </td>
<td >11755</td>
<td >56
</td>
<td>NA</td><td>NA</td></tr>
<tr style="font-style:italic">
<td>Heat.beta</td>
<td>1</td>
<td></td><td></td><td></td><td></td><td></td><td></td></tr>
<tr >
<td>Scale[1]</td>
<td>11.71 </td>
<td>(9.515, 14.27)</td>
<td >1.027  </td>
<td >11444</td>
<td >59
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>tn93:pi[A]</td>
<td>0.3071 </td>
<td>(0.2823, 0.3318)</td>
<td >6.696  </td>
<td >1755</td>
<td >74
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>tn93:pi[C]</td>
<td>0.2203 </td>
<td>(0.2027, 0.2381)</td>
<td >6.094  </td>
<td >1929</td>
<td >125
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>tn93:pi[G]</td>
<td>0.1897 </td>
<td>(0.1728, 0.2077)</td>
<td >6.783  </td>
<td >1732</td>
<td >292
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>tn93:pi[T]</td>
<td>0.2829 </td>
<td>(0.2588, 0.3057)</td>
<td >9.538  </td>
<td >1232</td>
<td >345
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>tn93:kappaPyr</td>
<td>2.826 </td>
<td>(2.488, 3.168)</td>
<td >4.096  </td>
<td >2869</td>
<td >172
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>tn93:kappaPur</td>
<td>2.083 </td>
<td>(1.834, 2.359)</td>
<td >4.765  </td>
<td >2466</td>
<td >118
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>rs07:mean_length</td>
<td>8.725 </td>
<td>(5.515, 12.67)</td>
<td >3.206  </td>
<td >3666</td>
<td >98
</td>
<td>NA</td><td>NA</td></tr>
<tr >
<td>rs07:log_rate</td>
<td>-5.427 </td>
<td>(-5.859, -5.04)</td>
<td >21.63  </td>
<td >543</td>
<td >79
</td>
<td>NA</td><td>NA</td></tr>
</table>
<h2><a class="anchor" name="topology"></a>Phylogeny Distribution</h2>
<img src="c-levels.svg" class="r_floating_picture svg-image" width="35%" /><img src="c50-tree.svg" class="floating_picture svg-image" width="25%" /><table>
<tr><td>Partition support: <a href="consensus">Summary</a></td><td><a href="partitions.bs">Across chains</a></td></tr>
</table>
<table>
<tr><td>50% consensus</td><td><a href="c50.tree">Newick</a> (<a href="c50.PP.tree">+PP</a>)</td><td><a href="c50-tree.pdf">PDF</a></td><td><a href="c50-tree.svg">SVG</a></td><td></td><td></td><td></td><td></td></tr><tr><td>66% consensus</td><td><a href="c66.tree">Newick</a> (<a href="c66.PP.tree">+PP</a>)</td><td><a href="c66-tree.pdf">PDF</a></td><td><a href="c66-tree.svg">SVG</a></td><td></td><td></td><td></td><td></td></tr><tr><td>80% consensus</td><td><a href="c80.tree">Newick</a> (<a href="c80.PP.tree">+PP</a>)</td><td><a href="c80-tree.pdf">PDF</a></td><td><a href="c80-tree.svg">SVG</a></td><td></td><td></td><td></td><td></td></tr><tr><td>90% consensus</td><td><a href="c90.tree">Newick</a> (<a href="c90.PP.tree">+PP</a>)</td><td><a href="c90-tree.pdf">PDF</a></td><td><a href="c90-tree.svg">SVG</a></td><td></td><td></td><td></td><td></td></tr><tr><td>95% consensus</td><td><a href="c95.tree">Newick</a> (<a href="c95.PP.tree">+PP</a>)</td><td><a href="c95-tree.pdf">PDF</a></td><td><a href="c95-tree.svg">SVG</a></td><td></td><td></td><td></td><td></td></tr><tr><td>99% consensus</td><td><a href="c99.tree">Newick</a> (<a href="c99.PP.tree">+PP</a>)</td><td><a href="c99-tree.pdf">PDF</a></td><td><a href="c99-tree.svg">SVG</a></td><td></td><td></td><td></td><td></td></tr><tr><td>100% consensus</td><td><a href="c100.tree">Newick</a> (<a href="c100.PP.tree">+PP</a>)</td><td><a href="c100-tree.pdf">PDF</a></td><td><a href="c100-tree.svg">SVG</a></td><td></td><td></td><td></td><td></td></tr><tr><td>MAP</td><td><a href="MAP.tree">Newick</a> (<a href="MAP.PP.tree">+PP</a>)</td><td><a href="MAP-tree.pdf">PDF</a></td><td><a href="MAP-tree.svg">SVG</a></td><td></td><td></td><td></td><td></td></tr><tr><td>greedy</td><td><a href="greedy.tree">Newick</a> (<a href="greedy.PP.tree">+PP</a>)</td><td><a href="greedy-tree.pdf">PDF</a></td><td><a href="greedy-tree.svg">SVG</a></td><td></td><td></td><td></td><td></td></tr></table>
<h2 class="clear"><a class="anchor" name="alignment"></a>Alignment Distribution</h2>
<h3>Partition 1</h3>
<table>
<tr>
<th></th>
<th></th>
<th></th>
<th title="Comparison of this alignment (top) to the WPD alignment (bottom)">Diff</th>
<th></th>
<th style="padding-right:0.5em;padding-left:0.5em" title="Percent identity of the most dissimilar sequences">Min. %identity</th>
<th style="padding-right:0.5em;padding-left:0.5em" title="Number of columns in the alignment"># Sites</th>
<th style="padding-right:0.5em;padding-left:0.5em" title="Number of invariant columns">Constant</th>
<th title="Number of parsiomny-informative columns.">Informative</th>
</tr>
<tr>
<td>Initial</td>
<td><a href="C1.P1.initial.fasta">FASTA</a></td>
<td><a href="C1.P1.initial.html">HTML</a></td>
<td><a href="C1.P1.initial-diff.html">Diff</a></td>
<td></td>
<td style="text-align: center">29.8%</td>
<td style="text-align: center">364</td>
<td style="text-align: center">26 (7.14%)</td>
<td style="text-align: center">295 (81%)</td>
</tr>
<tr>
<td>Best (WPD)</td>
<td><a href="P1-max.fasta">FASTA</a></td>
<td><a href="P1-max.html">HTML</a></td>
<td></td>
<td><a href="P1-max-AU.html">AU</a></td>
<td style="text-align: center">26.5%</td>
<td style="text-align: center">401</td>
<td style="text-align: center">27 (6.73%)</td>
<td style="text-align: center">328 (81.8%)</td>
</tr>
</table>
<h2><a class="anchor" name="mixing"></a>Mixing</h2>
<table style="width:100%;clear:both"><tr><td style="vertical-align:top"><ol>
</ol>
<p><b>Statistics:</b></p><table class="backlit2"><tr><td><b>scalar burnin</b></td><td>689</td></tr><tr><td><b>scalar ESS</b></td><td>68.84</td></tr><tr><td><b title="Effective Sample Size for bit-vectors of partition support (smallest)">topological ESS</b></td><td>15.450</td></tr><tr><td><b title="Average Standard Deviation of Split Frequencies">ASDSF</b></td><td>NA</td></tr><tr><td><b title="Maximum Standard Deviation of Split Frequencies">MSDSF</b></td><td>NA</td></tr><tr><td><b>PSRF CI80%</b></td><td>NA</td></tr><tr><td><b>PSRF RCF</b></td><td>NA</td></tr></table>
</td><td><img src="c50.SRQ.png" alt="SRQ plot for supprt of 50% consensus tree."/><img src="partitions.SRQ.png" alt="SRQ plot for support of each partition."/></td></tr></table><table style="width:100%;clear:both"><tr><td style="width:40%;vertical-align:top"><h4 style="text-align:center">Projection of RF distances for 1 chain (<a href="https://doi.org/10.1080/10635150590946961">Hillis et al 2005</a>)</h4><img src="tree1.svg" class="svg-image" width="90%" /><a href='tree-3D1-1.points.html'>3D</a></td><td style="width:40%;vertical-align:top"><h4 style="text-align:center">Variation of split PPs across chains (<a href="https://doi.org/10.1080/10635150600812544">Beiko et al 2006</a>)</h4><br/><br/><br/><br/><p>Not generated: multiple chains needed.</p></td></tr></table><br/><hr/><br/>
<h2><a class="anchor" name="analysis"></a>Analysis</h2>
<p><b>command line</b>: bali-phy data/refcov89_ORF8_aln.fasta</br>
<b>directory</b>: /home/russell/Projects/CoronaARG</br>
<b>version</b>: 3.4.1
</p><table class="backlit2 center">
<tr><th>chain #</th><th>burnin</th><th>subsample</th><th>Iterations (after burnin)</th><th>subdirectory</th></tr>
<tr>
  <td>1</td>
  <td>1306</td>
  <td>1</td>
  <td>11754</td>
  <td>refcov89_ORF8_aln-1</td>
</tr>
</table>
<br/>
<table style="width:100%">
<tr>
  <td>P(data|M) = -11282.582    +- 0.494
</td>
  <td>Complete sample: 13059
 topologies</td>
  <td>95% Bayesian credible interval: 12406 topologies</td>
</tr>
</table>
<h2 style="clear:both"><a class="anchor" name="models"></a>Model and priors</h2>
<h3 style="clear:both"><a class="anchor" name="tree"></a>Tree (+priors)</h3>
<table>
<tr><td class="modelname">topology</td><td>~ uniform on tree topologies</td></tr><tr><td class="modelname">branch lengths</td><td>~ iid[num_branches[T],gamma[0.5,div[2,num_branches[T]]]]</td></tr></table>
<h3 style="clear:both"><a class="anchor" name="smodel"></a>Substitution model (+priors)</h3>
<table class="models">
<tr><td><a name="S1" class="anchor"></a><span class="modelname">S1</span></td><td>=</td><td>
tn93
<table class="model">
<tr><td>tn93:kappaPur</td><td>~</td><td>log_normal[log[2],0.25]
</td></tr>
<tr><td>tn93:kappaPyr</td><td>~</td><td>log_normal[log[2],0.25]
</td></tr>
<tr><td>tn93:pi</td><td>~</td><td>dirichlet_on[letters[@a],1]
</td></tr>
</table></td></tr>
</table>
<h3 style="clear:both"><a class="anchor" name="imodel"></a>Indel model (+priors)</h3>
<table class="models">
<tr><td><a name="I1" class="anchor"></a><span class="modelname">I1</span></td><td>=</td><td>
rs07
<table class="model">
<tr><td>rs07:log_rate</td><td>~</td><td>laplace[-4,0.707]
</td></tr>
<tr><td>rs07:mean_length</td><td>~</td><td>exponential[10,1]
</td></tr>
</table></td></tr>
</table>
<h3 style="clear:both"><a class="anchor" name="scales"></a>Scales (+priors)</h3>
<table class="models">
<tr><td><a name="scale1" class="anchor"></a><span class="modelname">scale1</span></td><td>~</td><td>
gamma[0.5,2]
</td></tr>
</table>
  </div>
  </body>
</html>
